src/terralib.lua

```diff
@@ -3195,6 +3213,18 @@ function typecheck(topexp,luaenv,simultaneousdefinitions)
                 local br = s.branches:map(checkcondbranch)
                 local els = (s.orelse and checkblock(s.orelse))
                 return s:copy{ branches = br, orelse = els }
+            elseif s:is "switchstat" then
+                local cond = checkexpintegral(s.condition)
+                --local br = s.cases:map(checkcasebranch)
+                --local br = terralib.newlist()
+                --local stack = terralib.newlist{s.cases}
```

Comment:
Can we remove these comments if they're not needed?

Reply:
Oh, yeah. Sorry, forgot about those.